### DISCUSSION TITLE
kernel_4.1.38--ADv1.patch.xz download error making 7581

### DISCUSSION THREAD

**Discussion by ableeker** [Freetz-NG/freetz-ng#1056] [Allgemein] [answered]
Since 3 weeks, when I try to `make` the 7851 image, I get the following error:

```
--2024-10-20 15:58:52--  https://raw.githubusercontent.com/Freetz-NG/dl-mirror/master/kernel_4.1.38--ADv1.patch.xz
Resolving raw.githubusercontent.com... 2606:50c0:8002::154, 2606:50c0:8001::154, 2606:50c0:8000::154, ...
Connecting to raw.githubusercontent.com|2606:50c0:8002::154|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2024-10-20 15:58:52 ERROR 404: Not Found.

Download failed - "https://raw.githubusercontent.com/Freetz-NG/dl-mirror/master/kernel_4.1.38--ADv1.patch.xz"  ->  error code 8
make[1]: *** [make/kernel/kernel.mk:43: dl/kernel_4.1.38--ADv1.patch.xz] Error 1
make: *** [Makefile:47: envira] Terminated
```

Before that I could `make` images for the 7581successfully.

**Comment by fda77** [comment]
you should do a clean checkout

**Reply by ableeker** [reply]
I've done that. I deleted the folder, and downloaded the latest version of the code with `git clone`. Then I used `make menuconfig`to set up freetz-ng, and I only changed the hardware (7581), and the version of the firmware (international). Then I used `make` to create the image. The result was the same however, I got the same error. It looks like a file is missing.

**Comment by aus-sua** [comment]
Ich habe gerade eben einfach 'mal nur die gesamte url aus dem Error:
> `Download failed - "https://raw.githubusercontent.com/Freetz-NG/dl-mirror/master/kernel_4.1.38--ADv1.patch.xz"  ->  error code 8`

in den Browser geworfen.

Der sagt dann damit ebenfalls nur:
---> `404: Not Found`
an (wie ein einfaches curl darauf auch...). :-(

Die Datei scheint so also wirklich nicht hinter der url zu liegen?!
Mir fallen da auf die Schnelle nur:
- falscher Dateiname (der Pfad an sich sollte doch stimmen?!) oder 
- Datei nicht extern freigegeben (gibt es das in githubusercontent.com?)
 
ein.

---------

When entering the error-URL, the browser also only returns
---> `404: Not Found`
The file is probably not behind the URL?

**Comment by fda77** [comment]
Ja der Dateiname ist schon falsch, zwischen den `--` steht normal was. Es wird wohl an 73d5b848252dab0a230ca4a077ff5456e4533ccb liegen

Du kannst hier schauen welche Dateien es gibt: https://github.com/Freetz-NG/dl-mirror

Das Problem von  7581/7582 ist dass die AVM sourcen komplett Schrott und unbenutzbar sind. AVM hat da irgend was neues gemacht wazu ich keine Lust hab mich damit zu beschäftigen. Daher werden diese Sourcen nicht genutzt.
Gleiches gilt für 7510+1200ax 7.3

**Comment by fda77** [comment, marked as answer]
* 6365ddff0

**Reply by ableeker** [reply]
Thanks, great job! That did the trick all right, freetz-ng build successfully first try after this commit.

You were right, it seems that file should have been called `kernel_4.1.38-7581_07.18-ADv1.patch.xz)`. That one is available on the server.

**Reply by fda77** [reply]
sry, ich tested it some weeks ago only in no-compile mode and did not get this error while removing the avm-diff
